crates/chain/src/keychain/txout_index.rs

```diff
@@ -133,8 +125,11 @@ impl<K> KeychainTxOutIndex<K> {
     }
 }
 
+/// Methods that are *re-exposed* from the internal [`SpkTxOutIndex`].
 impl<K: Clone + Ord + Debug> KeychainTxOutIndex<K> {
     /// Return a reference to the internal [`SpkTxOutIndex`].
+    ///
+    /// **WARNING:** The internal index will contain lookahead spks.
```

Comment:
nit: maybe we should expand a little bit on stored vs revealed vs lookahead.

Reply:
I think we should do this on the main type. It already does this a bit but it could be clearer.

Reply:
83e7b7ec402bef27c3f83876f346ec2abd23eff1